test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,552 @@
+        expect(err.errors[0].path).to.include('uniqueName');
+        expect(err.errors[0].message).to.equal('custom unique error message');
+      });
+
+      it('should handle multiple unique messages correctly', async () => {
+        class Book extends Model<InferAttributes<Book>, InferCreationAttributes<Book>> {
+          declare uniqueName1: string;
+          declare uniqueName2: string;
+        }
+
+        Book.init({
+          uniqueName1: {
+            type: DataTypes.STRING,
+            unique: {
+              name: 'uniqueName1',
+              msg: 'custom unique error message 1',
+            },
+          },
+          uniqueName2: {
+            type: DataTypes.STRING,
+            unique: {
+              name: 'uniqueName2',
+              msg: 'custom unique error message 2',
+            },
+          },
+        }, { sequelize });
+
+        const records = [
+          { uniqueName1: 'unique name one', uniqueName2: 'unique name one' },
+          { uniqueName1: 'unique name one', uniqueName2: 'this is ok' },
+          { uniqueName1: 'this is ok', uniqueName2: 'unique name one' },
+        ];
+        await Book.sync({ force: true });
+        const instance = await Book.create(records[0]);
+        expect(instance).to.be.ok;
+        const err0 = await expect(Book.create(records[1])).to.be.rejected;
+        expect(err0 instanceof ValidationError);
+        expect(err0.errors).to.have.length(1);
+        expect(err0.errors[0].path).to.include('uniqueName1');
+        expect(err0.errors[0].message).to.equal('custom unique error message 1');
+
+        const err = await expect(Book.create(records[2])).to.be.rejected;
+        expect(err instanceof ValidationError);
+        expect(err.errors).to.have.length(1);
+        expect(err.errors[0].path).to.include('uniqueName2');
+        expect(err.errors[0].message).to.equal('custom unique error message 2');
+      });
+    }
+  });
+
+  describe('#create', async () => {
+    describe('generic', async () => {
+      disableDatabaseResetForSuite();
+      const vars = beforeAll2(async () => {
+        class Project extends Model<InferAttributes<Project>, InferCreationAttributes<Project>> {
+          static associations: {
+            users: HasMany<Project, Task>,
+          };
+
+          declare name: CreationOptional<string>;
+
+          declare setTask: HasManyAddAssociationMixin<Task, number>;
+        }
+
+        Project.init({
+          name: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            defaultValue: 'unknown',
+            validate: {
+              isIn: [['unknown', 'hello', 'test']],
+            },
+          },
+        }, { sequelize });
+
+        class Task extends Model<InferAttributes<Task>, InferCreationAttributes<Task>> {
+          static associations: {
+            users: BelongsTo<Task, Project>,
+          };
+
+          declare ProjectId?: CreationOptional<number | null>;
```

Comment:
It's already optional because it's nullable

```suggestion
          declare ProjectId?: number | null;
```